lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp

```diff
@@ -5069,6 +5069,31 @@ void SwiftASTContext::PrintDiagnostics(DiagnosticManager &diagnostic_manager,
 
 void SwiftASTContext::ModulesDidLoad(ModuleList &module_list) {
   ClearModuleDependentCaches();
+  TargetSP target_sp = GetTarget().lock();
+  if (!target_sp)
+    return;
+  bool use_all_compiler_flags = target_sp->GetUseAllCompilerFlags();
+  unsigned num_images = module_list.GetSize();
+  for (size_t mi = 0; mi != num_images; ++mi) {
+    std::vector<std::string> module_search_paths;
+    std::vector<std::pair<std::string, bool>> framework_search_paths;
+    std::vector<std::string> extra_clang_args;
+    lldb::ModuleSP module_sp = module_list.GetModuleAtIndex(mi);
+
+    ProcessModule(module_sp, m_description, use_all_compiler_flags, *target_sp,
+                  module_search_paths, framework_search_paths,
+                  extra_clang_args);
+    if (use_all_compiler_flags && !extra_clang_args.empty()) {
+      // We cannot reconfigure ClangImporter after its
+      // creation. Instead poison the SwiftASTContext so it gets
+      // recreated.
+      m_fatal_errors.SetErrorStringWithFormat(
```

Comment:
Would it be worth adding an API on SwiftASTContext (ClearClangImporter or something) to do this?  I worry that having a fatal error pending on the context could get you into trouble if some other subsystem checks for errors but isn't prepared to reconstitute the context.

Reply:
The only way to clear clangimporter is to create a new Swift compiler instance (= a new SwiftASTContext). The recreation is transparently handled by Target's GetScratchContext function.